services/poeditor/poeditor.service.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { nonNegativeInteger } = require('../validators')
+const { coveragePercentage } = require('../color-formatters')
+const { BaseJsonService } = require('..')
+
+const documentation = `
+  <p>
+    You must specify the read-only API token from the POEditor account to which the project belongs.
+  </p>
+  <p>
+    As per <a href="https://poeditor.com/docs/api">the POEditor API documentation</a>,
+    <q>all requests to the API must contain the parameter api_token. You can get this key from your POEditor account.
+    You'll find it in <a href="https://poeditor.com/account/api">My Account > API Access</a>.</q>
+  </p>
+`
+
+const schema = Joi.object({
+  response: Joi.object({
+    code: nonNegativeInteger,
+    message: Joi.string().required(),
+  }).required(),
+  result: Joi.object({
+    languages: Joi.array()
+      .items({
+        name: Joi.string().required(),
+        code: Joi.string().required(),
+        percentage: Joi.number()
+          .min(0)
+          .max(100)
+          .required(),
+      })
+      .required(),
+  }),
+}).required()
+
+const queryParamSchema = Joi.object({
+  token: Joi.string().required(),
+}).required()
+
+module.exports = class POEditor extends BaseJsonService {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'poeditor',
+      pattern: 'progress/:projectId/:languageCode',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'POEditor',
+        namedParams: { projectId: '323337', languageCode: 'fr' },
+        queryParams: { token: 'abc123def456' },
+        staticPreview: this.render({
+          code: 200,
+          message: 'OK',
+          language: { percentage: 93, code: 'fr', name: 'French' },
+        }),
+        keywords: ['l10n'],
+        documentation,
+      },
+    ]
+  }
+
+  static render({ code, message, language }) {
+    const color = 'red'
+
+    if (code !== 200) {
+      return { message, color }
+    }
+
+    if (language === undefined) {
+      return { message: `Language not in project`, color }
+    }
```

Comment:
If we need to render an error response, can we do this by throwing an `InvalidResponse()` exception rather than manually rendering the error badge.

Reply:
Done.